File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/script/env/ItemScriptEnv.java
package net.bestia.zoneserver.script.env;

import net.bestia.model.geometry.Point;

import java.util.Map;

public class ItemScriptEnv extends ScriptEnv {

	private final Point targetPosition;
	private final long userId;
	private final long targetId;

	public ItemScriptEnv(long userId, long targetId
  ) {
		this(userId, targetId, null);
	}

	public ItemScriptEnv(long userId, Point targetPosition) {
		this(userId, 0, targetPosition);
	}

	public ItemScriptEnv(
			long userId,
			long targetId,
			Point targetPosition) {

		this.userId = userId;
		this.targetId = targetId;
		this.targetPosition = targetPosition;
	}

  @Override
  protected void customEnvironmentSetup(Map<String, Object> bindings) {
    bindings.put("SELF", userId);
    bindings.put("TARGET_ENTITY", targetId);
    bindings.put("TARGET_POSITION", targetPosition);
  }
}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/script/env/ItemScriptEnv.kt
package net.bestia.zoneserver.script.env

import net.bestia.model.geometry.Vec3

class ItemScriptEnv private constructor(
    private val userId: Long,
    private val targetId: Long?,
    private val targetPosition: Vec3? = null
) : ScriptEnv {

  constructor(userId: Long, targetId: Long) : this(userId, targetId, null)
  constructor(userId: Long, targetPosition: Vec3) : this(userId, null, targetPosition)

  override fun setupEnvironment(bindings: MutableMap<String, Any?>) {
    bindings["SELF"] = userId
    bindings["TARGET_ENTITY"] = targetId
    bindings["TARGET_POSITION"] = targetPosition
  }
}
